core/services/ping/pingmanager.py

```diff
@@ -1,38 +1,68 @@
+import asyncio
 import logging
-from typing import Dict, List
+from typing import Any, Dict, List
 
 from serial import Serial
 
 from ping1d_driver import Ping1DDriver
 from ping360_driver import Ping360Driver
 from pingdriver import PingDriver
-from pingutils import PingDeviceDescriptor, PingType
+from pingutils import PingDeviceDescriptor, PingType, udp_port_is_in_use
 
 
 class PingManager:
     def __init__(self) -> None:
         self.drivers: Dict[PingDeviceDescriptor, PingDriver] = {}
-        self.ping1d_current_port: int = 9090
-        self.ping360_current_port: int = 9092
+        self.ping1d_base_port: int = 9090
+        self.ping360_base_port: int = 9092
 
     def stop_driver_at_port(self, port: Serial) -> None:
         """Stops the driver instance running for port "port" """
-        ping_at_port = list(filter(lambda ping: ping.port == port, self.drivers.keys()))
+        ping_at_port = [ping for ping in self.drivers if ping.port == port]
         if ping_at_port:
             self.drivers[ping_at_port[0]].stop()
+            del self.drivers[ping_at_port[0]]
 
-    def launch_driver_instance(self, ping: PingDeviceDescriptor) -> None:
+    async def register_ethernet_ping360(self, ping: PingDeviceDescriptor) -> None:
+        if ping not in self.drivers:
+            self.drivers[ping] = Ping360Driver(ping, None)
+
+    async def find_next_port(self, base_port: int, direction: int) -> int:
+        """
+        Find the next unused udp port, starts on base_port and increments/decrements
+        accordingly to direction
+        """
+        port = base_port
+        while udp_port_is_in_use(port):
+            port += direction
```

Comment:
Makes it clear it could also be used with a step-size greater than 1.
```suggestion
    async def find_next_port(self, base_port: int, step: int) -> int:
        """
        Finds the next unused UDP port.
        Starts at 'base_port' and increments/decrements by 'step'.
        """
        port = base_port
        while udp_port_is_in_use(port):
            port += step
```